### PR TITLE
Buildx in OSS

### PR DESCRIPTION
**Pull request by Bslabe123**
# Description

Docker now uses buildx to build images.

# Context

Many of the arch specific issues encountered in enterprise were fixed by using the buildx plugin which has support for multi-platform builds. This is a safe change since we don't modify any existing docker build behavior.

Changes can be verified by running `make docker -B`

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/7384

### CODE REVIEW

**Comment by Bslabe123** [comment]
/kick https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=e248080d-8a5b-4773-a5bf-47260ed48668

**Review by sam-heilbron — COMMENTED** [review]
Really like the heavily focused PR!

File: Makefile [hunk 1]
```diff
@@ -114,12 +114,20 @@ LDFLAGS := "-X github.com/solo-io/gloo/pkg/version.Version=$(VERSION)"
...
-	PLATFORM=--platform=linux/amd64
+# if `GO_ARCH` is set, then it will keep its value. Else, it will be changed based off the machine's host architecture.
+# if the machines architecture is set to arm64 then we want to set the appropriate values, else we only support amd64
+IS_ARM_MACHINE := $(or	$(filter $(UNAME_M), arm64), $(filter $(UNAME_M), aarch64))
+ifneq ($(IS_ARM_MACHINE), )
+	PLATFORM := --platform=linux/amd64
+	ifneq ($(GOARCH), amd64)
+		GOARCH := arm64
+		PLATFORM := --platform=linux/arm64
+	endif
```
Comment (kdorosh) [inline Makefile:125]: can't we simplify this to roughly:
```
         PLATFORM := --platform=linux/$(GOARCH)
```

Comment (kdorosh) [reply Makefile:125]: leaving the goarch check, but not duplicating platform part

**Review by irateswami — APPROVED** [review]

**Review by ianmacclancy — APPROVED** [review]
Tested locally - gtg

**Pull request merged**